openedx/core/lib/block_cache/block_structure.py

```diff
@@ -0,0 +1,586 @@
+"""
+Module with family of classes for block structures.
+    BlockStructure - responsible for block existence and relations.
+    BlockStructureBlockData - responsible for block & transformer data.
+    BlockStructureXBlockData - responsible for xBlock data.
+
+The following internal data structures are implemented:
+    _BlockRelations - Data structure for a single block's relations.
+    _BlockData - Data structure for a single block's data.
+"""
+# pylint: disable=protected-access
+from collections import defaultdict
+from logging import getLogger
+
+from openedx.core.lib.graph_traversals import traverse_topologically, traverse_post_order
+
+from .exceptions import TransformerException
+
+
+logger = getLogger(__name__)  # pylint: disable=C0103
+
+
+# A dictionary key value for storing a transformer's version number.
+TRANSFORMER_VERSION_KEY = '_version'
+
+
+class _BlockRelations(object):
+    """
+    Data structure to encapsulate parents and children relationship for
+    a single block.
+    """
+    def __init__(self):
+
+        # List of usage keys of this block's parents.
+        # list [UsageKey]
+        self.parents = []
+
+        # List of usage keys of this block's children.
+        # list [UsageKey]
```

Comment:
These comments (L34-35 and L38-39) should probably be part of the class docstring, so the documentation better explains how the object works.  E.g.

    """
    Data structure to encapsulate relationships for a single block.
    
    Attributes:
        parents - a list of usage keys for this block's parent blocks
        children - a list of usage keys for this block's child blocks
    """